include/contractor/contractor_dijkstra.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef OSRM_CONTRACTOR_DIJKSTRA_HPP
+#define OSRM_CONTRACTOR_DIJKSTRA_HPP
+
+#include "contractor/contractor_graph.hpp"
+#include "contractor/contractor_heap.hpp"
+
+#include <cstddef>
+
+namespace osrm
+{
+namespace contractor
+{
+
+// allow access to the heap itself, add Dijkstra functionality on top
+class ContractorDijkstra : public ContractorHeap
```

Comment:
This is kind of a inheritance misuse IMHO. An algorithm and a data structure don't share the same concept. It would be cleaner to make this a member and wrap access if needed.